libsolidity/formal/CHC.h

```diff
@@ -79,44 +79,76 @@ class CHC: public SMTEncoder
 	void unknownFunctionCall(FunctionCall const& _funCall);
 	//@}
 
+	struct IdCompare
```

Comment:
By the way I created https://github.com/ethereum/solidity/issues/8319 for tracking to collect all of these and maybe share the logic across the compiler.

Reply:
Nice!